### PR TITLE
Update keras-nlp landing page for 0.4 release

### PR DESCRIPTION
**Pull request by jbischof**
Closes https://github.com/keras-team/keras-nlp/issues/502.

### CODE REVIEW

**Comment by jbischof** [comment]
Cannot commit until #1159 is merged.

File: templates/keras_nlp/index.md [hunk 1]
```diff
@@ -2,20 +2,25 @@
...
-KerasNLP is also new and growing! If you are interested in contributing, please
+KerasNLP is a natural language processing library that supports users through
+their entire development cycle. Our workflows are built from modular components 
+that have state-of-the-art preset weights and architectures when used 
+out-of-the-box and are easily customizable when more control is needed. We 
+emphasize in-graph computation for all workflows so that developers can expect 
+easy productionization using the Tensorflow ecosystem.
+
+This library is an extension of the core Keras API; all high-level modules are 
+[`Layers`](https://keras.io/api/layers/) or
```
Comment (mattdangerw) [inline templates/keras_nlp/index.md:13]: I think you can just do `/api/layers/` and ditto for other links on this page. Then they will stay correct for local previews as well.

File: templates/keras_nlp/index.md [hunk 2]
```diff
@@ -39,64 +44,72 @@ check out our
...
+with our components. While we continue with pre-release `0.y.z` development, we
+may break compatibility at any time and APIs should not be consider stable.
+
+## Disclaimer
+
+KerasNLP provides access to pre-trained models via the `keras_nlp.models` API.
+These pre-trained models are provided on an "as is" basis, without warranties
+or conditions of any kind. The following underlying models are provided by third
+parties, and subject to separate licenses:
+DistilBERT, RoBERTa, XLM-RoBERTa, GPT-2.
```
Comment (mattdangerw) [inline templates/keras_nlp/index.md:111]: Let's add deberta here.

**Review by mattdangerw — APPROVED** [review]
Just a few minor changes.

File: templates/getting_started/ecosystem.md [hunk 3]
```diff
@@ -17,13 +17,12 @@ KerasTuner is an easy-to-use, scalable hyperparameter optimization framework tha
...
-flexibly iterate on your task. Engineers working in applied NLP can leverage the
-library to assemble training and inference pipelines that are both
-state-of-the-art and production-grade. KerasNLP is maintained directly by the
-Keras team.
+KerasNLP is a natural language processing library that supports users through
+their entire development cycle. Our workflows are built from modular components 
+that have state-of-the-art preset weights and architectures when used 
+out-of-the-box and are easily customizable when more control is needed. We 
+emphasize in-graph computation for all workflows so that developers can expect 
+easy productionization using the Tensorflow ecosystem.
```
Comment (fchollet) [inline templates/getting_started/ecosystem.md:25]: Fix capitalization of TensorFlow

File: templates/keras_nlp/index.md [hunk 4]
```diff
@@ -2,20 +2,24 @@
...
-added to core Keras, but that receive the same level of polish as the rest of
-the Keras API.
-
-KerasNLP is also new and growing! If you are interested in contributing, please
+KerasNLP is a natural language processing library that supports users through
+their entire development cycle. Our workflows are built from modular components 
+that have state-of-the-art preset weights and architectures when used 
+out-of-the-box and are easily customizable when more control is needed. We 
+emphasize in-graph computation for all workflows so that developers can expect 
+easy productionization using the Tensorflow ecosystem.
```
Comment (fchollet) [inline templates/keras_nlp/index.md:10]: Same

File: templates/keras_nlp/index.md [hunk 5]
```diff
@@ -39,64 +43,72 @@ check out our
...
+[`keras_nlp.models`](/api/keras_nlp/models/) API:
 
 ```python
 import keras_nlp
-import tensorflow as tf
 from tensorflow import keras
+import tensorflow_datasets as tfds
+
+# Use mixed precision for optimal performance
+keras.mixed_precision.set_global_policy('mixed_float16')
```
Comment (fchollet) [inline templates/keras_nlp/index.md:70]: Not sure we should enable mixed precision in a starter code thing. It's orthogonal to everything else. In a `/examples/` code example (configured to run on GPU in Colab) it's more clearly useful.

**Review by fchollet — APPROVED** [review]
LGTM

**Comment by fchollet** [comment]
I'm merging, I won't actually update the live website until the whole nlp subsite is ready though.

**Pull request merged**